internal/discoverer/nacos.go

```diff
@@ -0,0 +1,335 @@
+package discoverer
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/nacos-group/nacos-sdk-go/model"
+
+	"github.com/api7/apisix-seed/internal/utils"
+
+	"github.com/nacos-group/nacos-sdk-go/clients"
+	"github.com/nacos-group/nacos-sdk-go/vo"
+
+	"github.com/api7/apisix-seed/internal/conf"
+	"github.com/api7/apisix-seed/internal/core/comm"
+	"github.com/nacos-group/nacos-sdk-go/clients/naming_client"
+	"github.com/nacos-group/nacos-sdk-go/common/constant"
+)
+
+func init() {
+	Discoveries["nacos"] = NewNacosDiscoverer
+}
+
+type NacosDiscoverer struct {
+	weight int
+	// nacos client default config
+	ClientConfig constant.ClientConfig
+	// nacos server configs
+	ServerConfigs []constant.ServerConfig
+	// nacos naming clients
+	namingClients map[string]naming_client.INamingClient
+
+	params map[string]*vo.SubscribeParam
+	cache  map[string]Service
+	mu     sync.Mutex
+
+	watchCh chan *comm.Watch
+}
+
+func NewNacosDiscoverer(disConfig interface{}) (Discoverer, error) {
+	config := disConfig.(*conf.Nacos)
+	timeout := config.Timeout
+
+	clientConfig := constant.ClientConfig{
+		// compatible with past timeout configurations
+		TimeoutMs:           uint64(timeout.Connect + timeout.Read + timeout.Send),
+		NamespaceId:         config.Namespace,
+		Username:            config.Username,
+		Password:            config.Password,
+		NotLoadCacheAtStart: true,
+	}
+
+	serverConfigs := make([]constant.ServerConfig, len(config.Host))
+	for i, host := range config.Host {
+		schemaEnd, ipEnd := strings.Index(host, "://"), strings.LastIndex(host, ":")
```

Comment:
Is it possible to use some library to parse URL?

Reply:
Now I'm using `net/url` to parse URL.